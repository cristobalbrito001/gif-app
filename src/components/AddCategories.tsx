import React,{ useState,FC, FormEvent, ChangeEvent }   from "react";

interface AddCategoryProps {
    onNewCategorie: (category: string) => void;
  }
const AddCategory :FC<AddCategoryProps> = ({onNewCategorie})=>{

    const [inputValue,setInputValue] = useState('')
    const onInputChange = (event:ChangeEvent<HTMLInputElement>) =>{
        const newCategorie = event.target.value
        setInputValue(newCategorie)
    }

    const onSubmit = (e: FormEvent<HTMLFormElement>) =>{
        e.preventDefault();
        if(inputValue.trim().length  === 0){
            return
        }
        onNewCategorie(inputValue);
        setInputValue('')
    }
   return( 
    <form onSubmit={  onSubmit } aria-label="form">
        <h3>
            <input
                type="text"
                placeholder="buscar..."
                value={inputValue}
                onChange={ onInputChange }
            />
        </h3>
    </form>
    )
}

export{
    AddCategory
}