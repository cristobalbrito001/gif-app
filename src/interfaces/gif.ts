interface OriginalImage {
    url: string;
  }
  
  interface ImageType {
    original: OriginalImage;
  }
  
  interface gif {
    id: string;
    images: ImageType;
    title: string;
  }
export type{
    gif
}