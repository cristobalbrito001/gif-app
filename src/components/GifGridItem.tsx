import { gif } from '../interfaces/gif'
import React from 'react'
const GifItem =({images  ,title}:gif) =>{
    console.log(images)
    return(
        <div className="card">
            <img src={images.original.url} alt={title} />
            <p>{title}</p>
        </div>
    )
}
export{
    GifItem
}