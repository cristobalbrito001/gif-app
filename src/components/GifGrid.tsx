import React from 'react'
import Swal from 'sweetalert2';
import { GifItem } from './GifGridItem';
import { useFetchGif } from '../hooks/useFetchGif'

interface GifGridProps {
    category: string;
  }

const GifGrid: React.FC<GifGridProps> = ( { category }) =>{

    const {images, isLoading} = useFetchGif( category )

    if (!isLoading && images.length > 0) { 
        Swal.fire({
            toast: true,
            position: 'top-end',
            icon: 'success',
            title: 'desfruta tus gif bro',
            showConfirmButton: false,
            timer: 2000,
            timerProgressBar: true,
          });
    }
    if(images.length < 1){
        Swal.fire({
            title: "no hay imagenes",
            text: "wtf",
            icon: "error"
          });
    }
    return(
        <>
            <h3> {category} </h3>
            <div className='card-grid'>
                {images.map((image) => (
                   <GifItem key={image.id} {...image}/>
                ))}
            </div>
        </>
    )
}

export{
    GifGrid
}