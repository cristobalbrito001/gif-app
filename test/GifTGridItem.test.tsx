import{ gif } from '../../gif-app/src/interfaces/gif'
import { GifItem } from '../src/components/GifGridItem'
import { screen, render } from '@testing-library/react';
import React from 'react'
import { ExportClause } from './../node_modules/escodegen/escodegen';
describe('',()=>{
    const git :gif ={
        id:'asasasa',
        images:{
            original:{
                url:'https://media0.giphy.com/media/v1.Y2lkPWE4ZjI5ZjVidDczN3l3MmRycjA5d20zejYwY3VkaTVpdjFrNGRxZmVyYWF2N2NwNiZlcD12MV9naWZzX3NlYXJjaCZjdD1n/eATt3b9TKBfB2l0z4G/giphy.gif'
            }
        },
        title:'bleach'
    }
    const getContainer = render(<GifItem key={git.id} {...git} />)
    test('',()=>{

        expect(getContainer).toMatchSnapshot()
    })

    test('',()=>{
        render(<GifItem key={git.id} {...git} />)
        const testScreen = screen.getByRole('img');
        expect(testScreen.getAttribute('src')).toContain(git.images.original.url)
        expect(testScreen.getAttribute('alt')).toContain(git.title)
    })

    test('', ()=>{
        render(<GifItem key={git.id} {...git} />)
        expect(screen.getByText(git.title)).toBeTruthy();
    })
})