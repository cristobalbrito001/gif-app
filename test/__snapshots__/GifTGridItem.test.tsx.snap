// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`  1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card"
      >
        <img
          alt="bleach"
          src="https://media0.giphy.com/media/v1.Y2lkPWE4ZjI5ZjVidDczN3l3MmRycjA5d20zejYwY3VkaTVpdjFrNGRxZmVyYWF2N2NwNiZlcD12MV9naWZzX3NlYXJjaCZjdD1n/eATt3b9TKBfB2l0z4G/giphy.gif"
        />
        <p>
          bleach
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card"
    >
      <img
        alt="bleach"
        src="https://media0.giphy.com/media/v1.Y2lkPWE4ZjI5ZjVidDczN3l3MmRycjA5d20zejYwY3VkaTVpdjFrNGRxZmVyYWF2N2NwNiZlcD12MV9naWZzX3NlYXJjaCZjdD1n/eATt3b9TKBfB2l0z4G/giphy.gif"
      />
      <p>
        bleach
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
